lighthouse-cli/bin.ts

```diff
@@ -250,14 +250,30 @@ function handleError(err: LighthouseError) {
 }
 
 function runLighthouse(url: string,
-                       flags: {port: number, skipAutolaunch: boolean, selectChrome: boolean, output: any, outputPath: string, view: boolean},
+                       flags: {port: number, skipAutolaunch: boolean, selectChrome: boolean, output: any,
+                         outputPath: string, view: boolean, saveArtifacts: boolean, saveAssets: boolean},
                        config: Object): Promise<undefined> {
 
   let chromeLauncher: ChromeLauncher;
   return initPort(flags)
     .then(() => getDebuggableChrome(flags))
     .then(chrome => chromeLauncher = chrome)
     .then(() => lighthouse(url, flags, config))
+    .then((results: Results) => {
+      // delte artifacts from result so reports won't include artifacts.
+      const artifacts = results.artifacts;
+      results.artifacts = undefined;
+      if (flags.saveArtifacts) {
+        assetSaver.saveArtifacts(artifacts);
+      }
+      if (flags.saveAssets) {
+        return assetSaver.saveAssets({url: results.url}, artifacts).then(() => results);
+      }
+      return results;
+    })
+    .then((results: Results) => {
+      return results;
```

Comment:
Oops. Sorry, wasn't careful enough when checking after rebase. Didn't notice that. Already fixed.